app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,12 @@
+module Archivable
+  extend ActiveSupport::Concern
+
+  included do
+    scope :archived, -> { where('archived_at is not null') }
+    scope :published, -> { where(archived_at: nil) }
+  end
+
+  def archive
+    update_attributes(archived_at: Time.now)
```

Comment:
I thought that too.. but that is not thread safe. It might not matter too much, but I like to draw attention to all the unthreadsafe things we do, to develop awareness of the issue.